core/modules/replica/FileIngestApp.cc

```diff
@@ -0,0 +1,291 @@
+    ).required(
+        "table",
+        "The name of a table to be ingested.",
+        _file.tableName
+    ).required(
+        "type",
+        "The type of a table to be ingested. Allowed options: 'P' for contributions"
+        " into partitioned tables, and 'R' for contributions into the regular tables.",
+        _file.tableType
+    ).required(
+        "infile",
+        "A path to an input file to be sent to the worker.",
+        _file.inFileName
+    );
+
+    parser().command(
+        "FILE-LIST"
+    ).description(
+        "The batch ingest option. A list of files to be ingested will be read from"
+        " a file.  Each line of the file specifies a destination of the ingest and"
+        " the name name of a file to ingest via the following space-separated fields:"
+        " <worker-host> <worker-port> <transaction-id> <table> {'P'|'R'} <infile>."
+        " Where 'P' is for the partitioned (chunked) table contributions, and 'R' is"
+        " for the regular tables contributions. Input files for the partitioned tables"
+        " are expected to have the following names: 'chunk_<num>.txt' or"
+        " 'chunk_<num>_overlap.txt'. The files will be ingested sequentially."
+    ).required(
+        "file-list",
+        "The name of a file with ingest specifications. If the file name is set to '-'"
+        " then the specifications will be read from the Standard Input Stream",
+        _fileListName
+    );
+}
+
+int FileIngestApp::runImpl() {
+    string const context = "FileIngestApp::" + string(__func__) + "  ";
+
+    list<FileIngestSpec> files;
+    if (_command == "FILE") {
+        files.push_back(_file);
+    } else if (_command == "FILE-LIST") {
+        files = _readFileList();
+    } else {
+        throw invalid_argument(
+                context
+                + "Unsupported loading method " + _command);
+    }
+    for (auto&& file: files) {
+        _ingest(file);
+    }
+    return 0;
+}
+
+
+list<FileIngestApp::FileIngestSpec> FileIngestApp::_readFileList() const {
+    string const context = "FileIngestApp::" + string(__func__) + "  ";
+
+    list<FileIngestSpec> files;
+    int lineNum = 0;
+
+    for (auto&& line: util::File::getLines(_fileListName)) {
+        ++lineNum;
+
+        // Fields to be parsed:
+        // <worker-host> <worker-port> <transaction-id> <table> <type> <infile>
+        regex const re("^(.+)[ ]+([0-9]+)[ ]+([0-9]+)[ ]+(.+)[ ]+(P|R)[ ]+(.+)$", regex::extended);
+        smatch match;
+        if (not regex_search(line, match, re) or match.size() != 7) {
+            throw invalid_argument(
+                    context
+                    + "Failed to parse the content of file " + _fileListName + ":" + to_string(lineNum)
+                    + ", expected fields: <worker-host> <worker-port> <transaction-id> <table> <type> <infile>.");
+        }
+
+        FileIngestSpec file;
+        file.workerHost = match[1].str();
+
+        auto const workerPort = stoul(match[2].str());
```

Comment:
I guess you're sure these will not throw, based on the regex. But the regex is a little obscure to read, and could be changed by some unwitting future developer (probably named Nate). It might be safer to try/catch these for safety?  Up to you tho.